### PR TITLE
Make OpenSSL on windows only build for current build type

### PR DESCRIPTION
**Pull request by ThadHouse**
Previously, both debug and release were built anytime openssl was built, even if only 1 or the other was needed. This fixes that, so only the proper target is built.

### CODE REVIEW

File: .azure/templates/build-config-user.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
 
   - task: Cache@2
     inputs:
-      key:  '"${{ parameters.platform }}_${{ parameters.arch }}_${{ parameters.tls }}_${{ parameters.extraName }}_5" | .gitmodules'
+      key:  '"${{ parameters.platform }}_${{ parameters.arch }}_${{ parameters.tls }}_${{ parameters.extraName }}_${{ parameters.config }}_5" | .gitmodules'
```
Comment (nibanks) [inline .azure/templates/build-config-user.yml:49]: You could reset the number at the end now, right?

Comment (ThadHouse) [reply .azure/templates/build-config-user.yml:49]: Yeah I can.

File: CMakeLists.txt [hunk 2]
```diff
@@ -403,31 +403,48 @@ if(QUIC_TLS STREQUAL "openssl")
...
+        add_library(OpenSSL_Debug STATIC IMPORTED)
+        set_target_properties(OpenSSL_Debug PROPERTIES
+            IMPORTED_LOCATION_DEBUG ${OPENSSL_DIR}/debug/lib/libssl${CMAKE_STATIC_LIBRARY_SUFFIX}
+            IMPORTED_LINK_INTERFACE_LANGUAGES_DEBUG "C")
+
+        target_include_directories(OpenSSL_Debug INTERFACE
+                ${OPENSSL_DIR}/debug/include)
+        target_link_libraries(OpenSSL_Debug INTERFACE OpenSSL_Crypto_Debug)
+
+
```
Comment (nibanks) [inline CMakeLists.txt:420]: nit: remove double new lines

**Review by nibanks — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -446,34 +446,30 @@ if(QUIC_TLS STREQUAL "openssl")
                 COMMAND if not exist \"${QUIC_BUILD_DIR}/submodules/openssl/debug\" mkdir \"${QUIC_BUILD_DIR}/submodules/openssl/debug\" 2> NUL)
             add_custom_target(mkdir_openssl_build_release
                 COMMAND if not exist \"${QUIC_BUILD_DIR}/submodules/openssl/release\" mkdir \"${QUIC_BUILD_DIR}/submodules/openssl/release\" 2> NUL)
+
             add_custom_command(
                 DEPENDS mkdir_openssl_build_debug
+
```
Comment (nibanks) [inline CMakeLists.txt:452]: I'd prefer not to have these extra new lines.

**Review by nibanks — APPROVED** [review]
Looks good. Just one nit, but I don't care too much if it's fixed.

**Pull request merged**